### ISSUE TITLE
Seperate build status for each job in a build matrix

### ISSUE THREAD

**Issue by chuckatkins** [appveyor/ci#1805]
Currently, the status reported to GitHub for a build is the combined pass/fail of all builds.  It would be a great feature to have each configuration in the build matrix post a separate status so that instead of a single `continuous-integration/appveyor/pr` status, you would get separate statuses for `ci/appveyor: vs2015` and `ci/appveyor: vs2017`, for example.  This is a really useful feature available in CircleCI now where each independent "job" get's it's own status message.

**Comment by preed** [comment]
As discussed in this question/support ticket https://help.appveyor.com/discussions/questions/16443-status-badge-per-matrix-build, it sounds like this is the feature we're describing a desire for as well (with slightly different language).

**Comment by chuckatkins** [comment]
Yes, I believe they are one in the same. I used "job" in this context to reference circle CI, which is how they refer to the individual build matrix entries (at least in circle CI 2.0), each of which gets its own GitHub status entry.

**Comment by IlyaFinkelshteyn** [comment]
@chuckatkins can you point to some existing open source commit status to see how this look like?

**Comment by chuckatkins** [comment]
Certainly!  An easy example to see this in is the ADIOS2 project (I'm one of the core developers).  You can see from one of the currently open PRs: https://github.com/ornladios/ADIOS2/pull/382 we're using travis to run code format tests with clang-format, circle-ci for linux tests where we use docker containers to run two jobs, one for gcc 4.8 and another for gcc7 + openmpi, and then appveyor for vs2015 and vs2017 tests.  You can see here how the two circle-ci builds show up as separate status entries while the appveyor build only shows up as one.

**Comment by IlyaFinkelshteyn** [comment]
Another comment from customer email:
Not sure whether it's possible at all, but it would be neat if each job in the matrix (or, better yet, some to-be-defined groups of jobs) could be shown as separate checks in GitHub pull requests; e.g. in case of our project, it would be neat if the Windows builds would show up as one check and the Unix build would show up as another one. (This is a feature we currently have because we're using a different CI service for the Unix builds.)

**Comment by IlyaFinkelshteyn** [comment]
Related issue: https://github.com/appveyor/ci/issues/1960

**Comment by spyoungtech** [comment]
This has a lot of value for us.


We have a repo with multiple solution files e.g. `src/aSolutionDir` and `src\anotherSolution`. Following [the advice here](https://help.appveyor.com/discussions/questions/833-target-multiple-solution-files) we do this via a build matrix.

We'd like a badge to report on the build status of each solution independently. To accomplish this, we have tried to configure multiple projects, which works*. The consequence, however, is that every commit to the repr triggers builds for *all* the projects, which is untenable.

Allowing status badges per job would alleviate this.

Alternatively (and somewhat unrelated to this issue) the ability to only run builds when changes to subdirectories are detected would also resolve our current frustrations. That way only relevant projects will build.

**Comment by IlyaFinkelshteyn** [comment]
@spyoungtech you resolve your current frustrations, you can use [commit filtering](https://www.appveyor.com/docs/how-to/filtering-commits/#commit-files-github-and-bitbucket-only) with custom YAML file [name](https://www.appveyor.com/docs/build-configuration/#yaml-file-alternative-naming) or [location](https://www.appveyor.com/docs/build-configuration/#alternative-yaml-file-location) for each project.

**Comment by spyoungtech** [comment]
@IlyaFinkelshteyn Thanks! This should work beautifully. Didn't notice the configuration was available in the UI because we had `ignore appveyor.yml` checked.

**Comment by IlyaFinkelshteyn** [comment]
@spyoungtech we introduce more and more concepts which are difficult to even expose in UI visually, so we recommend to use UI for initial configuration, but going forward export configuration to YAML and use it.

**Comment by CatGirlsAreLife** [comment]
After thinking about it I think it would be best to simply make it like:
\<Badge urls base with / without branch\>/\<system if not windows\>/\<image name/job name>?svg=true&\<svg text changes as needed\>

For the badges for readmes.

**Comment by IlyaFinkelshteyn** [comment]
@CatGirlsAreLife  We cannot make a specific promises but I believe it will be ready in new few months.

**Comment by Swader** [comment]
@IlyaFinkelshteyn it has been a few months, any progress you can share?

**Comment by zufuliu** [comment]
Any news?

**Comment by funjoker** [comment]
Is this still a thing?

**Comment by limdor** [comment]
What is the status here? Is there any plan that this will happen?

**Comment by axunonb** [comment]
[shield.io](https://shields.io/) seems support the requested feature on appveyor builds: `/appveyor/job/build/:user/:repo/:job/:branch`
